pkg/controller/baremetalhost/host_state_machine.go

```diff
@@ -42,6 +42,8 @@ func (hsm *hostStateMachine) handlers() map[metal3v1alpha1.ProvisioningState]sta
 		metal3v1alpha1.StateExternallyProvisioned: hsm.handleExternallyProvisioned,
 		metal3v1alpha1.StateMatchProfile:          hsm.handleMatchProfile,
 		metal3v1alpha1.StateReady:                 hsm.handleReady,
+		metal3v1alpha1.StatePreparing:             hsm.handlePreparing,
+		metal3v1alpha1.StatePrepared:              hsm.handlePrepared,
```

Comment:
How about ` Building` and `Preparing`?

Reply:
I split it because the state of `manual clean` in ironic is :
 `manageable`->`cleaning` or `clean wait`-> `manageable`(success) or `clean fail`(fail) .
If `manual clean` is successful, it will be `manageable`. But it is also `manageable` before `manual clean`. This will make me unable to judge whether it is before or after `manual clean`.
![avatar](https://docs.openstack.org/ironic/pike/_images/states.svg)
>[Manual cleaning](https://docs.openstack.org/ironic/latest/admin/cleaning.html) is typically used to handle long running, manual, or destructive tasks that an operator wishes to perform either before the first workload has been assigned to a node or between workloads. When initiating a manual clean, the operator specifies the cleaning steps to be performed. Manual cleaning can only be performed when a node is in the manageable state. Once the manual cleaning is finished, the node will be put in the manageable state again.

Reply:
Oh, i know! I was wrong. I can check `clean fail` in `Provisioning`.

Reply:
Nice, that's a clever solution.
The only downside I can think of here is that if an error occurs in cleaning then it will be reported as a provisioning error, which may make things harder to debug (or may be the correct way of thinking about it? I don't know).